src/ecMatching.ml

```diff
@@ -173,19 +177,31 @@ module Zipper = struct
     let (s, _) = split_at_cpos1 ~after:`No env cpos s in
     1 + List.length s
 
-  let zipper_at_nm_cpos1 (env : EcEnv.env) ((cp1, sub) : codepos1 * int) s zpr =
+  let zipper_at_nm_cpos1 (env : EcEnv.env) ((cp1, sub) : codepos1 * branch_select) s zpr =
     let (s1, i, s2) = find_by_cpos1 env cp1 s in
 
     match i.i_node, sub with
-    | Swhile (e, sw), 0 ->
+    | Swhile (e, sw), Cond true ->
         (ZWhile (e, ((s1, s2), zpr)), sw)
 
-    | Sif (e, ifs1, ifs2), 0 ->
+    | Sif (e, ifs1, ifs2), Cond true ->
         (ZIfThen (e, ((s1, s2), zpr), ifs2), ifs1)
 
-    | Sif (e, ifs1, ifs2), 1 ->
+    | Sif (e, ifs1, ifs2), Cond false ->
         (ZIfElse (e, ifs1, ((s1, s2), zpr)), ifs2)
 
+    | Smatch (e, bs), Match cn ->
+        let _, indt, _ = oget (EcEnv.Ty.get_top_decl e.e_ty env) in
+        let indt = oget (EcDecl.tydecl_as_datatype indt) in
+        let cnames = List.fst indt.tydt_ctors in
+        let ix, _ =
+          try List.findi (fun _ n -> EcSymbols.sym_equal cn n) cnames
```

Comment:
`findi` -> `find`

Reply:
I use `findi` to get the index of the element rather than the element itself.

Reply:
Oh, yes, sorry. I missed that it also returns the found element index.